### DISCUSSION TITLE
Feedback/Comments/General Discussion

### DISCUSSION THREAD

**Discussion by garungorp** [garungorp/MouseInjectorDolphinDuck#2] [General]
Please leave any comments, concerns, or questions. right here. If you have a suggestion on how something should be implemented for a certain game or if the game you know and love has been absolutely ruined by the addition of mouse support then let everybody know!

**Comment by EuanDewar** [comment]
Hey, thanks for all your work on this!

I'm having an issue. Attempting to play Darkwatch on PCSX2 nightly v1.7.4085 and bios SCPH-70012_BIOS_V12_USA_200. Everything loads fine, mouseinjector says it is detecting the game - but with the game hooked it doesn't actually work, despite saying it is hooked. Any idea what might be going on? Any details needed please do say!

**Reply by garungorp** [reply]
Darkwatch is slightly different from the other games in that to get it to work consistently I decided to use cheats to save the base addresses for variables into areas of the PS2 memory that are unused. There are also cheats that disable the auto-centering of the camera when falling and the aim-lock on enemies.

So you need to copy the file, '327053E8.pnach' from 'MouseInjectorDolphinDuck-latest\cheats' to the cheats folder in the PCSX2 directory. And then you need to enable cheats in PCSX2 by going to 'Settings->Emulation->Enable Cheats'

Let me know if this helps!

**Reply by EuanDewar** [reply]
That did work, thank you very much! Only issue now is that the cursor is still on-screen.

**Reply by EuanDewar** [reply]
Ignore me, pcsx2 has a setting for that. Again many thanks!

**Reply by garungorp** [reply]
Yeah you'll also have to disable the 'Double-Click Toggles Fullscreen' to avoid it constantly shrinking. I'll add a little note about these settings for PCSX2. Glad I could help!

**Comment by Akelaphobia** [comment]
Not trying to add more work to your plate, focus on the long list you've already got right now, but I'm just curious: down the road do you have plans to support RPCS3? There are a _lot_ of console-exclusive FPS/TPS' on there that'd be cool to see mouse support for. There's already a mouse injector for the emulator called KAMI, but it's been broken for a while and they've said recently that they're not actively developing it, sadly.

**Reply by garungorp** [reply]
Yeah I'd like to add support for RPCS3, xemu, xenia, redream, plus I've got a list of Sega Saturn games building and even some Sega Genesis games that I'd like to mess around with. I'm not going in any real order here so I just do whatever seems interesting. Also depends on the performance of the emulator. I haven't messed around too much with RPCS3 but looks like the games runs pretty well. So I may take a look at it soon if there's a game that really catches my interest and runs relatively smoothly.

**Reply by Akelaphobia** [reply]
Glad to hear that! When you do potentially add support for those, I'll give you a list of the exclusive FPS/TPS games on those systems. 

Dolphin and CEMU might also be worth a look, though really the only two on CEMU would be Devil's Third and ZombiU. I think every other shooter on that system was multiplatform (ZombiU was ported to PC, but it's missing some stuff from the WiiU version).

Wii on the other hand has quite a few good exclusive FPS', but at the moment they already thankfully work very well by just binding the Wiimote to your mouse since those games used a floating crosshair. Would be interesting though if you were able to lock the crosshair to the center of the screen and then enable mouse aim..

**Reply by Patrxgt** [reply]
> Yeah I'd like to add support for RPCS3, xemu, xenia, redream, plus I've got a list of Sega Saturn games building and even some Sega Genesis games that I'd like to mess around with.

Instead of redream you could add support to Flycast, since it's doing better job at emulating games and is FOSS.

**Comment by vini1264** [comment]
Oh hey, I would like to report some few issues regarding playing Darkwatch on PCSX2 with the injector, but I'm not really sure if it's something you can really fix. For example, if I raise PCSX2's overclock settings the mouse aiming stops working on the horse sections, and there's also times where after a cutscene or a level transition the mouse stops being captured for some brief seconds before it recovers.

**Reply by garungorp** [reply]
After some testing I was able to recreate the horse issue. I was only able to get it to happen if the game was started with the overclock already enabled. Seems like if you start the game at the normal clock speed and then increase it works fine. Only tried testing on the first horse level after restarting the emulator so not sure if it will break on subsequent levels.

The small pause on some levels is bound to happen just because of the way the addresses are being acquired. They flip between different values and the program checks to make sure a valid address is found and it may take a few cycles to find the right one resulting in a small pause usually at a level transition or restart. It's not a perfect solution but it's the most consistent one I was able to find for this game as there aren't any static addresses that hold pointers to the correct camera objects. And if they are there then they're most likely to be long pointer chains like the one used for the horse camera but that apparently breaks with overclocking so I'm not sure of the best solution. I may look at doing a cheat similar to the other ones that save the horse camera base in a known location. At least then it would most likely not break from overclocking.

I played through the whole game a few weeks ago and I don't remember any pauses that were too egregious but I did play on the normal clock speed and I'm not sure if raising it will make those pauses worse. So unfortunately that's how it is at the moment but if I come across a better solution in the future I'll make sure to come back to this one.

**Comment by ElTioRata** [comment]
I'm having issues with PCSX2, it doesn't seem to detect it so it closes. I guess it's because it's looking for the AVX2 executable instead of the SSE4 one (pcsx2-qtx64). There's a build that contains both instructions set (pcsx2-qt) so you might want to support that one instead.

**Reply by garungorp** [reply]
Yeah it just checks for one specific executable name which is the AVX2 version just because it was the version I happened to be using. I updated it to check for the SSE4 and Qt versions as well. I tried a few games and they seem to work the same but I'll hold off on adding them to the supported list until I've done a little more testing.

**Comment by TheKins** [comment]
Minor PCSX2 compatibility thoughts:

- 007: Agent Under Fire is not giving me the mouse-movement-during-cutscenes issues mentioned in the Readme
- Quake 3 Revolution, however, is - moving the mouse during in-engine cutscenes moves both the camera and your character. Since these cutscenes typically fly into the characters head for a transition to gameplay, this usually results in them ending with the camera awkward clipping all over the place.

Also: Thanks for this tool! It's off to a strong start and I hope it continues to grow.

**Reply by garungorp** [reply]
With Agent Under Fire it is usually the in-game cutscenes like one specifically I can think of is at the end of level 2 where the car jumps out of the building and drives down the street. The addresses used by the player's camera at that point are also used for the cinematic one so it will warp the camera. And for Quake it's a similar thing where the same camera is being used. Really they just need checks to see if the camera is moveable by the player. Not sure when I'll get to this but I've made notes of both.

Thanks for the feedback!

**Comment by dante3732** [comment]
Hello, please add reassigning buttons for functions or move them to the nampad, 4,5,6,7 interfere with assigning controlls.

**Reply by garungorp** [reply]
This a feature I've been wanting as well. Especially since I want to eventually try and add keyboard functionality to control in-game actions like assigning specific weapons in Quake to the number keys instead of having to scroll through them. I'll look at changing the the hotkeys to something else.

Thanks for the feedback!

**Reply by dante3732** [reply]
You can also make all the buttons work through a keyboard shortcut. For example cntrl+4, cntrl+5 and etc.

**Comment by quintenpm** [comment]
Hello! Im super excited to try shadow tower +kings field games, but im having trouble getting the mouse injector to detect the emulator running the game. At the top of the mouse injector terminal it says moue injector for duckstation - no game loaded. Ive tried running mouseinjector exe from the duckstation install folder but no luck

**Reply by garungorp** [reply]
There are a few things to check. What version of DuckStation do you have and are you using the standalone version? What BIOS are you using? Do the serials for the games match what is on the compatibility table? And have you downloaded the latest mouse injector release? Also the mouse injector executable doesn't have to be moved anywhere it should work from wherever you extracted it to.

I've tested it with a fresh download of DuckStation plus the mouse injector and it works so you might also try restoring the default settings in DuckStation as some setting might be causing it not to hook.

**Reply by quintenpm** [reply]
Resetting all default settings fixed it thank you! ill go back and enable all my settings one by one and figure out which one it doesnt like now. Also once I've gotten a decent control scheme ill post my setup here.

**Reply by garungorp** [reply]
Awesome! There was someone else who had the same issue but I couldn't narrow it down so it's good to hear that restoring the default settings fixes the issue.

**Comment by MrSunabouzu** [comment]
Just finished Echo Night using this, works perfect start to end! I tried to jump with spacebar a few times it felt natural enough lol

**Reply by garungorp** [reply]
Nice! Good to hear! I always think about trying to add stuff like jumping or crouching but then it just turns into making a mod of the game haha.

**Comment by robotrudy** [comment]
Hi there, trying to use this software with shadow tower on Duckstation and I can't figure out what to do with with the files!? I don't see an .exe or any other type of launchable file so I am curious as to how exactly I should even start the mouse injector...
cheers.

**Reply by Akelaphobia** [reply]
I'm going to take a wild guess and say you didn't download the latest zip file from the releases page (not the source code zip).

**Reply by robotrudy** [reply]
Yes, that is exactly what I did! thank you :^D

**Comment by dante3732** [comment]
By the way, tell us exactly how your injector works. Do you find coordinates in games and set the mouse to change them, or are you just emulating a stick on the mouse?

**Reply by garungorp** [reply]
It overwrites values in the game memory that correspond to the camera's angles with values calculated from the movement of the mouse. It's different from just using the mouse as input for the control stick because with that method, which is supported by PCSX2, you would still have controller based features like acceleration and snap-back applied to the mouse movements. With injection it just directly modifies the values without being adjusted.

Each game must be done separately as they have different addresses, types of variables used, overall structure, and different assisting features (they are different games after all). Generally I try to disable features like aim-lock, aim-assist that is rotational or slows down the camera, and auto-centering of the camera as they either interfere with the mouse injection or they make the aiming feel worse.

Also I am not the original creator of the project I am just maintaining it. The core of the program was made by the original creator and I have just modified it to work with other emulators and games. The core has basically gone unchanged since the original project.

Hope that helps!

**Comment by DfactorLongplays** [comment]
Seeing the Beetle PSX HW core being supported so quickly is definitely a pleasant surprise, thank you!
Just tested with the Steam version and it's working mighty fine. It would be a decent idea to have a thread where compatibility issues or lack thereof can be reported. I'd be definitely curious of doing some complete tests.

**Reply by garungorp** [reply]
Nice, good to hear that it's working fine! Yeah I can make a discussion for that because it would be nice to know if an update to a core or emulator breaks compatibility. I didn't do an extensive test with all the currently supported games for each core so it would be helpful. One issue I did find was with Medal of Honor Underground in that I didn't even know the game had an involuntary auto-center feature until I tried it with Beetle PSX and realized it was because the controller was set to digital.

Thanks for the feedback!

**Comment by RektAssley** [comment]
Heyo.
Having issues getting the mouse injector to work with Duckstation standalone client. Messed around with versions 0.1 5485 and 0.1 5437 and couldn't get Shadow Tower to get recognised by the mouseinjector. Could be the version I am using but am not able to acquire the specific version listed in the doc. If you have a link to it that would be amazing.
Works very well on PCSX2 with King's Field 4. Keen to see this project expand further!
Cheers!

**Reply by garungorp** [reply]
Tested it with 0.1-5485 and it's working so it must be some setting that is preventing it from hooking. The only one I know for sure at the moment is '**Settings->Console->Enable 8MB Ram (Dev Console)**' will prevent it from working if it is enabled. If that's not on then you should try '**Settings->Restore Defaults**' in DuckStation. A few other people have had this problem and that has fixed it.

Let me know if this helps!

**Comment by UserGit-2021** [comment]
Hello! Thank you for your hard work.
In the future, you can create a customization profile in mouseinjector. And you can add autoplay. And support Linux.

**Comment by OtterSlinger** [comment]
This is some truly great work, even games and systems I wouldn't have thought of being up for support, like SNES with Wild Guns, are getting lots of love from this project!

I thought I'd write in to ask for this, but I understand if it's just a complication of how the mouse injector works, would there be a chance you could make this compatible with Steam Input? It seems that any mouse movement done through that, for example setting up Gyro or using a controller's trackpad, doesn't get registered by the mouse injector. Again, understand if that's just outside the wheel house of the project. Oh, and this is for standard PC Steam use, not Steam Deck. From videos online, it seems like it works there with some Proton tinkering.

Also, for anyone looking for a workaround for this exact issue, ReWASD seems to work just fine doing the same thing, but that's a paid program.

**Reply by garungorp** [reply]
I didn't even know that Steam had this capability but I've played around with it a bit. I've tested it with Wild Guns and I see what you're saying. It seems to be because the injector only accepts input from one device and the Steam Input counts as a different one to the plugged in mouse. It uses whichever is input device is found first which usually is your primary mouse.

You can get it to work if you hook the game and then unplug your mouse. The injector will use the Steam Input as the mouse if it is the only other pointer device. Then you can plug your mouse back in and you'll have the opposite scenario where the mouse won't affect the game but the Steam Input controller will. It doesn't always seem to work so you might have to mess around with it.

A real fix will have to wait until some sort of GUI is implemented or I may be able to add a hotkey that loops through the detected input devices so the user can choose which input device to use instead of having to unplug the mouse.

Let me know if this workaround works for you too!

**Reply by OtterSlinger** [reply]
Unplugging and replugging my mouse works for me! Thank you for the work around, can't wait to make the most of the program using this!

**Comment by KarlGhartler** [comment]
Having some issues trying to get Wild Guns to hook. I've managed to get all the supported emulators (other than snes) working with it but I hadn't tried any SNES games til now. I saw your Wild Guns video so I had to try it out. I've tried it on the patched version of the ROM, as well as a version that combines the FastRom Patch from romhacking.net with the provided patch, and just the default game as well and none of them would detect. Not sure what i'm doing wrong unless Bsnes is only supported through Retroarch.

**Reply by garungorp** [reply]
The standalone BSNES is quite finicky with hooking as if you don't do everything in a specific order then it will not hook. Usually I would have to open BSNES, open the injector and start it, and immediately load the game for it to work but it was not very reliable. It would also generally stop working if you messed around too much with the menus or loaded save states. So I've removed it from the supported list for now although it technically will still work but the more reliable way to play it would be with RetroArch.

It should work with the original unpatched version, with the FastROM, and it works with and without the supplied patch as that one just affects the gameplay.

Unfortunately SNES is in a similar situation to N64 in that there doesn't seem to be a best choice when it comes to using the injector. They either don't have the option map the mouse buttons, to hide the cursor, or they don't have controller profiles which is understandable because I don't think there is any real need to have those options for playing the games normally. I would like to have an option for SNES games besides RetroArch so eventually I'll try and take a look at something like Ares or Mesen.

**Reply by KarlGhartler** [reply]
Thanks, appreciate your work and the support. I wasn't expecting to care much about the 2d Era of mouse support but as I've recently discovered Fastrom and SA-1 hacks I've got a reinvigorated interest.

**Comment by nikivarvar** [comment]
Hello!
I'm not sure if this is the correct thread to post this, but I've been interested in trying to implement the thing that would account for the AC turning speed in Armored Core using what you have done so far, but I ended up realizing that I overestimated my abilities, considering that I have zero to none experience with this, while attempting to find the address that directly affects the apparent maximum velocity, at which camera can turn. Albeit with help of a modding community I now have what was needed _at least to extent how I understand the problem_. I'm still stuck on figuring out what to do with it :p

I'll just drop what was found and hope that will be of any help, I tried.
0x411B6 (2 byte) seems like the address in question, does exactly what I expect it to, set the multiplied value of whatever was before and camera goes zoom when rotating, I'm assuming it can be used for clamping the velocity but idk how one does that.

**Reply by garungorp** [reply]
Awesome! There are a few ways you could go about implementing something like this so it depends on what result you're looking for.

Right now it just directly sets the angle that the AC is facing and the camera follows so the higher the sensitivity and the quicker you move the mouse the faster it will turn. A solution that would play closer to the original game is to instead affect the rotation speed with the mouse and clamp it to the current max turning speed stat so it won't turn faster than is normally possible no matter how fast you move the mouse. Another one I'd like to see is the ability to rotate the camera irrespective of the facing direction and the AC will rotate at it's max speed until it also facing the same direction.

Eventually I'd like the option to pick between these different ways of controlling the camera from within the program. I'll try to take a look at this soon as I know quite a few people are interested in this sort of implementation. If I can get it working then I'll upload it as a separate zip.

**Reply by nikivarvar** [reply]
Oh, just realized that I forgot to add that the value on that address changes when turning speed stat changes, that can be defo used to the first solution you described.
Second one seems interesting, and honestly makes a lot of sense to have, you get pure mouse look for the camera, but not the crosshair. Looking forward to it!

**Reply by KarlGhartler** [reply]
2nd option sounds best to me. Its how games like War Thunder operate

**Comment by melbbm** [comment]
I have mingw installed and the paths set for it, but I run into an error when trying to compile. It says the main.o file is missing:
![image](https://user-images.githubusercontent.com/18742906/224442492-96e6c25b-7f11-4d9c-9246-3b66c4d2fda1.png)

**Reply by melbbm** [reply]
I figured it out. I forgot a forward slash at the end of of the MINGWDIR directory.

**Comment by IDrinkLava** [comment]
I finally got the mouse injector working in Black. Thank you so much for this. It practically feels like [a native PC port now](https://youtu.be/yS_SkWsmDSk) with 60FPS and higher FOV set in the cheat file:

gametitle=BLACK (SLUS_21376)
patch=1,EE,203BE83C,word,00000001
patch=1,EE,204BC150,word,00000001
patch=1,EE,204BD18C,word,00000001
patch=1,EE,204CA554,word,3FE38E39
patch=1,EE,204CA5F0,word,3fe00000 // FOV
patch=1,EE,204CA5F4,word,3FE38E39
patch=1,EE,204CA694,word,3FE38E39
patch=1,EE,205BC390,word,3F400000
patch=1,EE,1040DF74,extended,00000001 // 60 fps
patch=1,EE,205A8A9C,extended,3C888889 // speed

**Comment by KarlGhartler** [comment]
One thing I've been thinking about, is that although there are many games on Dreamcast and PSX, for example, that natively support mouse, I find a unique enjoyment in playing games using your program because I get the benefit of still enjoying controller vibration as tactile feedback that is normally missing in native mouse/PC games, while still enjoying the mouse precision of the PC FPS games I typically play. Better yet I can rest the controller on my chest and it feels like the gun is kicking when I fire, sort of like those transducer vests or buttkicker devices.

**Comment by RADIOACTIVEshrek** [comment]
I really quite like the emulator, but I do have one small issue; My CPU doesn't support AVX2, therefore preventing me from playing the PS2 games. I know most have a CPU that supports it, but I haven't had the opportunity to upgrade. So, is it possible to see the mouse injector for a SSE4.1 build of PCSX2? If not, I'll understand.

**Reply by garungorp** [reply]
It technically does recognize the SSE4 version already but I just haven't added it to the supported list as I haven't tested many games with it although I believe there shouldn't be many issues with it.

**Comment by OtterSlinger** [comment]
Would it be possible to add an option for Cursor-aiming for Goldeneye and the Timesplitters games? To clarify, the 1964 GEPD's mouse injector has an option to make it so aiming works like it does in the unmodified games, unlocking the cursor from the center of the screen and the camera, letting you aim anywhere in your vision, light-gun style, while still using your mouse. Would love to see that option come to the Timesplitters games, and any other games that borrow that feature from Goldeneye!

**Comment by UCB-Andy** [comment]
Played thru the entirety of King's Field 1 with the mouse injector. Works perfectly from beginning to end. Over a decade ago i gave up playing the game 'cause it felt incredibly clunky, but thanks to your work i was able to enjoy the game for the first time, so thanks for all your hard work!

**Reply by RADIOACTIVEshrek** [reply]
I do recommend trying the game again some day with the original controls; It is worth learning to play it that way, as once you do, the clunkiness can actually add to it. Not too dissimilar from being able to learn Armored Core's controls before the 4th entry, it can be rewarding to master and understand their somewhat clunky systems.
 Shadow Tower Abyss and Eternal Ring are the ones who really benefit the most from the mouse injector though, given their faster paced combat.

**Reply by UCB-Andy** [reply]
Really enjoyed my time with the game, but I have so many other new games i'd like to play before replaying KF1 again.

**Comment by MOW531** [comment]
I'm having an issue with cod finest hour where on the seventh level 'Operation Little Saturn' the camera keeps getting reset for some reason. Is this expected?

Sorry if this is the wrong place to ask this.

**Reply by garungorp** [reply]
Yeah it's because the method of looking for the base address for the tank's camera isn't very robust. I wasn't aware that there were still issues with it but I've made a change that should hopefully make it more consistent and I'll upload a version with the fix here soon.

**Reply by MOW531** [reply]
Thank you!

**Comment by MOW531** [comment]
Call of Duty 2: Big Red One unfortunately doesn't have an option to disable aim assist, do you know if there's a patch for it?

**Comment by izumistaz2** [comment]
can i get a little help setting this? i'm a noob when it comes to this kind of stuff, i want to play King's field and Armored core with mouse support but i don't know where to start or what to do. the instructions say to start Mouseinjector but i downloaded the files and theres no .exe, it is a separed program?

**Reply by melbbm** [reply]
Hey I had the same problem as you but eventually figured it out. What you have downloaded is the source code. To get an exe, you need to compile the source code. Here's how:

1. Install mingw. There's tutorials online for this.
2. Run CMD and use the 'cd' command to get to the source code's directory.
3. Once you're in the directory, enter this command: "mingw23-make.exe -f makefile"

This should build the exe. You might have to make a small edit to the makefile so that it can find where you installed mingw. A makefile is basically a file of instructions for building the program, you can edit it with notepad. Here's what I changed mine to:
`#Compiler directories
MINGWDIR = C:/Program Files (x86)/mingw64/bin/`

**Reply by garungorp** [reply]
It's not necessary to build it yourself as "MouseInjectorDolphinDuck_vX.zip" comes with it compiled. When you go to the main project page, on the right you'll see 'Releases' and that is where you'll find downloads for the program. You just need to download it and extract it. It should contain 2 folders, **_cheats_** and **_patches_**, and 3 files, _**README.md**_, _**LICENSE**_, and _**Mouse Injector.exe**_.  If you have an active anti-virus it may quarantine the executable after it's extracted so you may need to make an exception for it in your anti-virus software.

You can download the zip file containing the program here:

- https://github.com/garungorp/MouseInjectorDolphinDuck/releases/download/v7/MouseInjectorDolphinDuck_v7.zip

**Reply by izumistaz2** [reply]
oh god, thanks. I was very overwhelmed with all this, I have no idea how to use code.

**Reply by melbbm** [reply]
There was a compiled exe this whole time? 😅 One day I'll understand how to use this website lol

**Comment by izumistaz2** [comment]
Is it possible to add Beetle Psx and Pcsx rearmed to the list of supported emulators?

**Reply by garungorp** [reply]
If you mean the RetroArch cores then I've added both of them which should be available in the next update. But if you're talking about a standalone version then you'll have to let me know which one.

**Reply by izumistaz2** [reply]
ohh i'm talking about the Beetle psx and pcsx rearmed cores. my pc is very trashy and can't handle the other psx cores. if it has to be in a standalone emulator, how about epsxe?

**Comment by DfactorLongplays** [comment]
I went through the entirety of King's Field IV: The Ancient City (USA) with mouse aim and would like to report that the experience is consistent throughout and the game can be finished from start to beginning with mouse without the need to switch back to the gamepad.

The whole experience will be available for viewing [here](https://www.youtube.com/playlist?list=PLp1mWA5o0CfSthUzphUczqDdp2NU6gf6o).
Do you accept such testing results?

**Reply by garungorp** [reply]
Nice, good to hear it works all the way through. Feel free to post any results if you do a full playthrough.

**Comment by UserGit-2021** [comment]
Check out this project. It would be nice to merge without using mouseinjector, but straight out of the emulator box.
https://github.com/marinesciencedude/xenia-canary-mousehook

**Comment by bubertiben** [comment]
so i've been stuck on AC2 ever since the patch came out and i can say it works flawlessly (minus the replays but i usually skip them anyways so idc)
i really like not needing to use the optional part that lets you quick turn and just hitting a quick 180 with my mouse after an over boost, i might as well be playing armored core 4! lol

**Comment by TBE0027** [comment]
I'm trying to use it on Black for PCSX2 but for some reason it just doesn't hook

PCSX2 v.1.7.4404 nightly, SCPH 90001 Bios, Black (USA) SLES-21376

I even managed to get it working shortly one time, but now it doesn't anymore :(
![image](https://user-images.githubusercontent.com/18484077/233791895-e418c656-251c-491a-9013-fae7e1c277c2.png)

**Reply by garungorp** [reply]
I'd recommend clicking the 'Restore Defaults' button in the settings, restart the emulator and injector, and then test with the default settings . Some settings may prevent the injector from hooking and they haven't been exhaustively tested so this is the easiest thing to try as it's been tested working on the latest PCSX2 version.

**Reply by Distec00** [reply]
Reporting I have a similar problem. I can't get the injector to detect any game no matter what version of MouseInjector or PCSX2 I use.

I've tried tried rolling PCSX2 back to 1.7.416 with no luck, and none better when updating to the most recent version. Also doesn't matter if I use the avx2.exe or the regular one. Tried several games including Black, Agent Under Fire, and Timesplitters 2 - all a bust. Using app defaults every step of the way.

I've been playing the Metroid Prime series via PrimeHack with complete success lately, so I'm at a loss as to what could be the issue here.

**Reply by garungorp** [reply]
@Distec00 I'm not able to recreate the problem so I'm not sure of the issue but Timesplitters 2 not working makes sense since it isn't supported on PCSX2. I've tested Black on a fresh install of Windows 10 with the latest PCSX2 version and it works so I have to assume it's something else causing the issue. 

I'd recommend closing any extraneous programs and then try it. Anything like Steam, anti-virus software, anti-cheat software for games, and so on. Someone even claimed that it wouldn't hook while Zoom or VirtualBox were running so I would close any other open programs or services to see if they're interfering, making sure to close and restart PCSX2 and MouseInjector as well. I'd also try unplugging any extraneous USB devices as well.

Also I'd try it with a supported game on Dolphin or DuckStation to determine if the problem is just with PCSX2 or with all emulators.

Let me know if this helps!

**Reply by YourKiss** [reply]
I have found solution for me it is State "feature" of PCSX2
Try to delete all States of game and restard emulator with injector, and don't use Load\Save State feature

**Comment by phasermaniac83** [comment]
Hi! amazing project! Do you know if all this is compatible with lightguns (absolute coordinate mice)?
Also, Is wildguns playable with 2 mice?
Thanks!

**Reply by garungorp** [reply]
MouseInjector locks the cursor in place and uses relative mouse movement so my guess is that a lightgun that uses absolute coordinate would feel quite strange and erratic but I've never tried it with a setup like that so I'm unsure.

As for Wild Guns, at the moment it only supports one mouse but it is possible to add support for two in the future.

**Reply by phasermaniac83** [reply]
Okay, thanks

**Comment by ArhumMK** [comment]
Have you thought of switching over to RMG (Rosalie's Mupen GUI) for the n64 mouse injector patches? It is currently the best emulator for N64 as it's based off of mupen64plus and is standalone with an excellent GUI.

**Reply by garungorp** [reply]
It technically does work with RMG already even though it's not listed as supported. It's not listed because it hasn't had a lot of testing and, as far as I can tell, you cannot map mouse buttons in RMG. I agree it would be the best to use as it has the nice Qt UI and controller profiles but without being able to map mouse inputs from within the GUI, it doesn't seem viable to use at the moment.

**Comment by Red-tv141** [comment]
Thanks for all your work

**Comment by DfactorLongplays** [comment]
Greetings!
I went through the entirety of Eternal Ring (USA) with mouse injector, and same as with King's Field IV: The Ancient City (USA), the game is playable with mouse aim from start to finish.
The results of the testing for Eternal Ring are [here](https://www.youtube.com/playlist?list=PLp1mWA5o0CfQHYbT_WwfLNHI3Iyq0imtH).
While a complete playlist for King's Field IV is [here](https://www.youtube.com/playlist?list=PLp1mWA5o0CfSthUzphUczqDdp2NU6gf6o).

Thank you for your work, I'll also test second Shadow Tower if you ever make it compatible.

**Reply by garungorp** [reply]
Nice! They are long games so it's good to know they work all the way through.

**Comment by Swordman14** [comment]
Hello! I finished Alien trilogy with moouse injector and tested Disruptor. Both worked perfectly! However I cant get mouse injector to work with Codename Tenka. Mouse injector detects the game and the mouse locks in place, but it just doesnt work in game. I tried both duckstation and beetle hw
Quick update: Found what was causing it. Turns out, if you turn on widescreen in game, it stops working for some reason. ¿Could that be fixed?

**Reply by garungorp** [reply]
Good to know the games work for you!

I was unware of Codename Tenka problem but it was a simple change and it should work now. I've uploaded the hotfix under the [v12 release](https://github.com/garungorp/MouseInjectorDolphinDuck/releases/tag/v12).

**Comment by bubertiben** [comment]
ghost in the shell (ps2) feels really good with a mouse and keyboard, funny enough with it having like no animations for swapping and reloading, it makes it feel snappier

**Comment by simkas** [comment]
What's the status of Killzone support on PS2? I see that there seems to be a profile made for it, the injector does detect it and say that it has hooked in, but the mouse still doesn't seem to work in the game, it gets locked in place, but doesn't move the aim.

**Comment by daemonspudguy** [comment]
I tried to use this with Dolphin on Wine, and it didn't even detect that I was running Serious Sam Next Encounter (NTSC v1.1) on Dolphin.

**Comment by JohnKirk1337** [comment]
What am I doing wrong here? I've tried running with Armored Core (USA) SCUS-94182 but it doesn't hook. It just says " Mouse Injector for DuckStation - No game loaded". I've tried to restore to default settings, even deleted the config folder altogether. I even download alternative bios. Nothing worked.

> Duckstation: 0.1-5485-gc6a57273 (dev)
> Edition	Windows 11 Pro
> Version	22H2
> OS build	22621.1848

**Reply by yzia2000** [reply]
Might be the Windows 11. Do other games work?

**Comment by duig0** [comment]
this is awesome software, it has transformed resident evil gun survivor into a proper FPS now lol.

i had a thought though, could you use Opentracks mouse emulation with this to create headtracking?

i tried, and although the mouse hooks, and i can see the curser bound on the screen, it doesnt move ingame, the regular mouse does, but not the emulated mouse

i know this os outside the scope of your project

but perhaps you might have some insight in getting the emulated mouse working?

**Reply by KarlGhartler** [reply]
That would be crazy but what games would you use it on? I already use a freepie script to conver TRACKIR movement to analog stick movement in PCSX2 when I play Ace Combat 4 and 5 in first person cockpit mode and after setting up a curve profile to compensate for the controller deadzone it works almost perfectly. I just cant think of many games that have a freelook independent of the movement.

**Comment by izumistaz2** [comment]
did the project died?

**Comment by UserGit-2021** [comment]
> Please leave any comments, concerns, or questions. right here. If you have a suggestion on how something should be implemented for a certain game or if the game you know and love has been absolutely ruined by the addition of mouse support then let everybody know!

Please don't drop the project! Thanks for your hard work!

**Comment by BoardScribbles** [comment]
Hello,
First of all, your mouse injector is amazing, i played a lot of games whith it! It is far better than a pad for first person games.
But lately, in ran into one issue.

I am trying to play Alien Trilogy (Ps1), and whatever the emulator I use (Duckstation, Retroarch...) or the mouse injector version I use (I tried all versions from V4), the issue is the same:
When I turn ON the mouse injector, everything works perfectly fine, it hooks properly, i can use the mouselook, except there's a kind of drift going up slowly, taking about 10 seconds to do a full 360 revolution (the camera is literally doing slow loopings).
And when i dont move the mouse it stops about 30 seconds after, but when i move the mouse again, it also starts again.

I already tried the classic workarounds: Restart the computer, try a different mouse, unplug and plug the mouse... no luck so far.

Thanks for your attention. I’m looking forward to your reply.

**Reply by garungorp** [reply]
This game requires a couple cheats that are provided when you download a release. You need to import the cheat file from the cheats folder and enable them in DuckStation.

![cheat](https://github.com/garungorp/MouseInjectorDolphinDuck/assets/61608560/43f42715-1a25-45dd-af80-ce819fb208c6)

![import_cheat](https://github.com/garungorp/MouseInjectorDolphinDuck/assets/61608560/8ab62b43-403a-45b4-b057-3da93708700c)

**Reply by BoardScribbles** [reply]
Thanks a lot for the answer! It is definitely the best way to play this game!

**Comment by Lamphyris** [comment]
Played King's field IV with the mouse injector, game completely beat, no issues whatsoever.

**Comment by Fv40** [comment]


**Reply by daemonspudguy** [reply]
Scroll down to releases.

**Comment by Instandhaltung** [comment]
First of all, thanks for the injector. Awesome stuff! King's Field 1 (JAP) is the only one that doesnt work well for me. Sometimes it detects the game but doesn't apply mouse movement, often it doesn't detect the game at all. Out of 10 tries, it only worked once ramdomly. After a restart, it didn't work again, and I can't pinpoint why it suddently worked. Maybe another way of detecting the game would fix it?

Using standalone Duckstation 
Resetting settings didn't solve it.

**Comment by IDDQD1337** [comment]
I really hope for the support of N.O.V.A. Near Orbit Vanguard Alliance (PSP). This game requires mouse support

**Comment by yzia2000** [comment]
Has anyone tried this with windows 11? Wondering if it's my setup not working or the mouse injector itself doesn't work on windows 11.

**Comment by Mahim2000** [comment]
Can we please get a Linux port?

**Comment by UserGit-2021** [comment]
The project is dead. For those who want to develop this project further, here is a training video from this author.
https://youtu.be/LLtaIrMyQBc
